traffic_ops_ort/plugin_verifier/plugin_verifier.go

```diff
@@ -0,0 +1,287 @@
+					if !exists {
+						verified = verifyPluginConfigfile(param)
+						pluginParams[param] = verified
+					}
+					if !verified {
+						log.Errorf("the plugin config file '%s' on line '%d' or near continuation line '%d' is not available to the installed trafficserver.\n", param,
+							lineNumber, lineNumber)
+						pluginErrorCount++
+					}
+				}
+			}
+		}
+	} else { // process a line from plugin.config
+
+		// process a line from plugin.config
+		if strings.HasSuffix(fields[0], ".so") {
+			key := strings.TrimSpace(fields[0])
+			verified, exists = pluginChecks[key]
+			if !exists {
+				verified = verifyPlugin(key)
+				pluginChecks[key] = verified
+			}
+			if !verified {
+				log.Errorf("the plugin '%s' on line '%d' is not available to the the installed trafficserver.\n", key,
+					lineNumber)
+				pluginErrorCount++
+			}
+		}
+		// Check the arguments in a plugin.config file for possible plugin config files.
+		// Any plugin argument that ends in '.config | .cfg | .txt' are
+		// assumed to be configuration files and are checked that they
+		// exist in the filesystem at the absolute location in the name
+		// or relative to the ATS configuration files directory.
+		m := regexp.MustCompile(`^*(\.config|\.cfg|\.txt)+`)
+		for ii := 1; ii < length; ii++ {
+			param := strings.TrimSpace(fields[ii])
+			if m.MatchString(param) {
+				verified, exists = pluginParams[param]
+				if !exists {
+					verified = verifyPluginConfigfile(param)
+					pluginParams[param] = verified
+				}
+				if !verified {
+					log.Errorf("the plugin config file '%s' on line '%d' or near continuation line '%d' is not available to the installed trafficserver.\n", param,
+						lineNumber, lineNumber)
+					pluginErrorCount++
+				}
+			}
+		}
+	}
+	return pluginErrorCount
+}
+
+// returns 'filename' exists 'true' or 'false'
+func fileExists(filename string) bool {
+	log.Debugf("verifying plugin file at %s\n", filename)
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// read the names of all available plugins in the
+// installed trafficservers plugin directory.
+func loadAvailablePlugins() {
+	files, err := ioutil.ReadDir(cfg.TrafficServerPluginDir)
+	if err != nil {
+		log.Errorf("%v\n", err)
+		os.Exit(1)
+	}
+
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".so") {
+			log.Debugf("loaded plugin %s\n", file.Name())
+			atsPlugins[file.Name()] = 1
+		}
+	}
+}
+
+func verifyPluginConfigfile(filename string) bool {
+	if filepath.IsAbs(filename) {
+		return fileExists(filename)
+	} else {
+		return fileExists(cfg.TrafficServerConfigDir + "/" + filename)
+	}
+}
+
+// returns plugin is verified (filename exists), 'true' or 'false'
+func verifyPlugin(filename string) bool {
+
+	if !strings.HasSuffix(filename, ".so") {
+		return false
+	}
+
+	if filepath.IsAbs(filename) {
+		return fileExists(filename)
+	} else {
+		return fileExists(cfg.TrafficServerPluginDir + "/" + filename)
```

Comment:
same as above RE: path separator joining

Reply:
@ocket8888 I made the change.